src/fondant/pipeline/pipeline.py

```diff
@@ -181,6 +179,22 @@ def __init__(
 
         self.resources = resources or Resources()
 
+    @classmethod
+    def from_component_yaml(cls, path, **kwargs):
```

Comment:
this will be the entrypoint for reusable and custom components (the ones that the python_component does not cover)